### PR TITLE
Add service to environment variables

### PR DESCRIPTION
**Pull request by coilysiren**
## Context for reviewers

https://betagrantsgov.slack.com/archives/C05TSL64VUH/p1728501460077109

## Testing

```
  # module.service.aws_ecs_task_definition.app must be replaced
-/+ resource "aws_ecs_task_definition" "app" {
      ~ arn                      = "arn:aws:ecs:us-east-1:315341936575:task-definition/api-dev:186" -> (known after apply)
      ~ arn_without_revision     = "arn:aws:ecs:us-east-1:315341936575:task-definition/api-dev" -> (known after apply)
      ~ container_definitions    = jsonencode(
          ~ [
              ~ {
                  ~ environment            = [
                        # (5 unchanged elements hidden)
                        {
                            name  = "DB_USER"
                            value = "app"
                        },
                      + {
                          + name  = "ENVIRONMENT_NAME"         # <===========
                          + value = "dev"                      # <===========
                        },
                        {
                            name  = "PORT"
                            value = "8000"
                        },
                        # (1 unchanged element hidden)
                    ]
                  ~ linuxParameters        = {
                      ~ capabilities       = {
                          - add  = []
                            # (1 unchanged attribute hidden)
                        }
                        # (1 unchanged attribute hidden)
                    }
                  - mountPoints            = []
                    name                   = "api-dev"
                  ~ portMappings           = [
                      ~ {
                          - hostPort      = 8000
                          - protocol      = "tcp"
                            # (1 unchanged attribute hidden)
                        },
                    ]
                  - systemControls         = []
                  - volumesFrom            = []
                    # (8 unchanged attributes hidden)
                },
            ] # forces replacement
        )
      ~ id                       = "api-dev" -> (known after apply)
      ~ revision                 = 186 -> (known after apply)
      - tags                     = {} -> null
        # (12 unchanged attributes hidden)
    }

Plan: 1 to add, 3 to change, 1 to destroy.
```

### CODE REVIEW

File: infra/api/service/main.tf [hunk 1]
```diff
@@ -138,7 +138,7 @@ module "service" {
     }
   } : null
 
-  extra_environment_variables = local.service_config.extra_environment_variables
+  extra_environment_variables = merge(local.service_config.extra_environment_variables, { "ENVIRONMENT_NAME" : var.environment_name })
```
Comment (chouinar) [inline infra/api/service/main.tf:141]: Could we call the variable `ENVIRONMENT`? That's what we're expecting in the API code

**Review by mdragon — APPROVED** [review]
Approved

File: infra/api/service/main.tf (hunk 1 above)
Comment (coilysiren) [reply infra/api/service/main.tf:141]: @chouinar done!

**Review by chouinar — APPROVED** [review]
LGTM!

**Pull request merged**